scripts/obtain/dotnet-install.sh

```diff
@@ -75,16 +75,24 @@ get_legacy_os_name_from_platform() {
             echo "debian"
             return 0
             ;;
+        "debian.9")
+            echo "linux"
+            return 0
+            ;;
         "fedora.23")
             echo "fedora.23"
             return 0
             ;;
+        "fedora.24")
+            echo "fedora.24"
+            return 0
+            ;;
         "fedora.27")
             echo "fedora.27"
             return 0
             ;;
-        "fedora.24")
-            echo "fedora.24"
+        "fedora.28")
+            echo "linux"
```

Comment:
Why is it `fedora.27` for Fedora 27 but `linux` for Fedora 28? Shouldn't it be the same for both? Similar comment for `debian`/`linux`.

Reply:
Fedora27 and Debian8 came before the 1.1 portable CLI was built, so they have their own version-specific CLIs they can use. Fedora28 and Debian8 came after the 1.1 portable CLI so they can just use that instead of having their own.

Fedora27 and Debian8 could **probably** also use the portable CLI, but I haven't tested it and there's not any benefit to doing so when they have their own CLIs already that I already know work.

Reply:
Thanks for clarifying. This is a very confusing piece of logic :)

Reply:
@ianhays 
I do not think this is correct. You are changing the behavior for legacy assets. Legacy assets are distro-specific - I would expect the return for debian.9 to be "debian.9" to match the naming convection of the debian.9 distro-specific asset.

Reply:
@johnbeisner there aren't any legacy assets for Debian9 or Fedora28 yet, so adding a new entry shouldn't affect any existing scenarios, regardless of what is being echoed. 

Echoing Fedora.28 would result in an attempted download of `dotnet-dev-fedora.28-x64...` which does not exist. What does exist, however, is `dotnet-dev-linux-x64...`, so we echo `linux` for Fedora28 to use that one instead.

Reply:
@ianhays 
This should already work in the way you have described - see: get_current_os_name()
Fed28 and Deb9 should echo: "Linux" in this context. Fed28 and Deb9 should fail the call to get_legacy_os_name_from_platform() because "there aren't any legacy assets for Debian9 or Fedora28 yet" - there shouldn't be anything to fix here.

Reply:
> This should already work in the way you have described - see: get_current_os_name()

It does not.

```
dotnet-install: Calling: machine_has curl
dotnet-install: Calling: calculate_vars
dotnet-install: Calling: get_azure_channel_from_channel rel-1.0.0
dotnet-install: azure_channel=rel-1.0.0
dotnet-install: Calling: get_normalized_architecture_from_architecture <auto>
dotnet-install: Calling: get_machine_architecture
dotnet-install: Calling: get_normalized_architecture_from_architecture x64
dotnet-install: normalized_architecture=x64
dotnet-install: Calling: get_specific_version_from_version https://dotnetcli.azureedge.net/dotnet rel-1.0.0 x64 1.0.0-preview3-003223-6
dotnet-install: specific_version=1.0.0-preview3-003223-6
dotnet-install: Calling: construct_download_link https://dotnetcli.azureedge.net/dotnet rel-1.0.0 x64 1.0.0-preview3-003223-6
dotnet-install: Calling: get_current_os_name
dotnet_install: Error: OS name could not be detected: debian.9
```

> Fed28 and Deb9 should fail the call to get_legacy_os_name_from_platform() because "there aren't any legacy assets for Debian9 or Fedora28 yet" - there shouldn't be anything to fix here.

That's the whole point of this PR though, I'm trying to add the legacy assets.

Reply:
I pasted the wrong thing in the earlier comment. This is what the error is without adding entries to the legacy getter:

```
dotnet-install: Calling: machine_has curl
dotnet-install: Calling: calculate_vars
dotnet-install: Calling: get_normalized_architecture_from_architecture <auto>
dotnet-install: Calling: get_machine_architecture
dotnet-install: Calling: get_normalized_architecture_from_architecture x64
dotnet-install: normalized_architecture=x64
dotnet-install: Calling: get_specific_version_from_version https://dotnetcli.azureedge.net/dotnet LTS x64 1.0.0-preview3-003223
dotnet-install: specific_version=1.0.0-preview3-003223
dotnet-install: Calling: construct_download_link https://dotnetcli.azureedge.net/dotnet LTS x64 1.0.0-preview3-003223
dotnet-install: Calling: get_current_os_name
dotnet-install: Calling: get_linux_platform_name
dotnet-install: download_link=https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-sdk-1.0.0-preview3-003223-linux-x64.tar.gz
dotnet-install: Calling: construct_legacy_download_link https://dotnetcli.azureedge.net/dotnet LTS x64 1.0.0-preview3-003223
dotnet-install: Calling: get_legacy_os_name
dotnet-install: Calling: get_legacy_os_name_from_platform debian.9
dotnet-install: Distribution specific OS name and version could not be detected: UName = Linux
dotnet-install: Cound not construct a legacy_download_link; omitting...
dotnet-install: Calling: resolve_installation_path <auto>
dotnet-install: Calling: get_user_install_path
dotnet-install: resolve_installation_path: user_install_path=/git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux
dotnet-install: install_root=/git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux
dotnet-install: Calling: check_pre_reqs
dotnet-install: Calling: install_dotnet
dotnet-install: Calling: is_dotnet_package_installed /git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux sdk 1.0.0-preview3-003223
dotnet-install: Calling: combine_paths /git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux sdk
dotnet-install: combine_paths: root_path=/git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux
dotnet-install: combine_paths: child_path=sdk
dotnet-install: Calling: combine_paths /git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux/sdk 1.0.0-preview3-003223
dotnet-install: combine_paths: root_path=/git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux/sdk
dotnet-install: combine_paths: child_path=1.0.0-preview3-003223
dotnet-install: is_dotnet_package_installed: dotnet_package_path=/git/cse4/build_projects/dotnet-host-build/../../.dotnet_stage0/Linux/sdk/1.0.0-preview3-003223
dotnet-install: Zip path: /tmp/dotnet.NxSKtDNPC
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-sdk-1.0.0-preview3-003223-linux-x64.tar.gz
dotnet-install: Calling: download https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-sdk-1.0.0-preview3-003223-linux-x64.tar.gz /tmp/dotnet.NxSKtDNPC
dotnet-install: Calling: machine_has curl
dotnet-install: Calling: downloadcurl https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-sdk-1.0.0-preview3-003223-linux-x64.tar.gz /tmp/dotnet.NxSKtDNPC
curl: (22) The requested URL returned error: 404
dotnet-install: Curl download failed
dotnet-install: Download failed: https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-sdk-1.0.0-preview3-003223-linux-x64.tar.gz
dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-sdk-1.0.0-preview3-003223-linux-x64.tar.gz
dotnet_install: Error: Could not download .NET Core SDK version 1.0.0-preview3-003223
```

Reply:
@ianhays 
There are no Fed28 and Deb9 SDK assets @ 1.0.0-preview3-003223-6
Nor is there a Linux portable SDK asset - I cannot see how this drop will work for your purposes.

Try this:
1. Make your changes to get_legacy_os_name_from_platform() with this in mind:
Legacy assets are distro-specific - I would expect the return for debian.9 to be "debian.9" to match the naming convection of the debian.9 distro-specific asset.

2. Test the new script against a blob storage location where you have placed your bootstrapped Fed28 and Deb9 SDKs

This is how we recently added the other 3 distros - let me know if I can assist.

Reply:
> There are no Fed28 and Deb9 SDK assets @ 1.0.0-preview3-003223-6
Nor is there a Linux portable SDK asset 

That is not correct. There is a Linux portable sdk for this version using the legacy package naming convention. It's right here: https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-dev-linux-x64.1.0.0-preview3-003223.tar.gz 

> I cannot see how this drop will work for your purposes.

This PR makes the legacy package restoration step work for Fed28 and Deb9 by having them pull down the portable package. I've already tested these changes in core-setup by using a local copy of the modified dotnet-install script and it **does** work without fail. 

> Legacy assets are distro-specific - I would expect the return for debian.9 to be "debian.9" to match the naming convection of the debian.9 distro-specific asset.
> Test the new script against a blob storage location where you have placed your bootstrapped Fed28 and Deb9 SDKs

Your assertion that legacy assets have to be distro-specific is no longer correct. See the above package. There is no longer a need for a bootstrap fed28 and Deb9 specific CLI.

Reply:
@ianhays 
"I could rename the bootstrap CLI" - where is the bootstrap CLI for Fed28 and Deb9? I cannot detect it's presence in 1.0.0-preview3-003223-6 nor 1.0.0-preview3-003223

Reply:
There is no longer a need for a bootstrap fed28 and Deb9 specific CLI for that version. I am using: https://dotnetcli.azureedge.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-dev-linux-x64.1.0.0-preview3-003223.tar.gz

Reply:
@johnbeisner and I discussed this in person and since this script is customer-facing I'm going to add a dotnet-install script to core-setup to get the portable 1.1 linux for the new distros.